src/gz.rs

```diff
@@ -862,6 +863,17 @@ impl Header {
     pub fn mtime(&self) -> u32 {
         self.mtime
     }
+
+    /// Returns the most recent modification time as a datetime if time stamp is available.
```

Comment:
Could you expand the documentation here as well to explain when `None` is returned and to detail how the `mtime` field is interpreted?